tuf/api/metadata.py

```diff
@@ -453,15 +452,25 @@ def verify_signature(
 
             signed_serializer = CanonicalJSONSerializer()
 
-        if not sslib_keys.verify_signature(
-            self.to_securesystemslib_key(),
-            signature.to_dict(),
-            signed_serializer.serialize(metadata.signed),
-        ):
+        try:
+            if not sslib_keys.verify_signature(
+                self.to_securesystemslib_key(),
+                signature.to_dict(),
+                signed_serializer.serialize(metadata.signed),
+            ):
+                raise exceptions.UnsignedMetadataError(
+                    f"Failed to verify {self.id} signature",
+                    metadata.signed,
+                )
+        except (
+            sslib_exceptions.CryptoError,
+            sslib_exceptions.FormatError,
+            sslib_exceptions.UnsupportedAlgorithmError,
+        ) as e:
             raise exceptions.UnsignedMetadataError(
-                f"Failed to verify {self.id} signature for metadata",
+                f"Failed to verify {self.id} signature",
```

Comment:
Can we use "e" to add these descriptive messages that are promised in the description? 
Or I should assume that this is to be added since this is still a draft?

Reply:
Well, I said "error message" but mean the whole traceback that includes e.g. the securesystemslib error if key is malformed or something like that.